### PR TITLE
Epoch setup random source update

### PR DESCRIPTION
**Pull request by tarakby**
This PR is part of improving the security of the epoch source of randomness after updating the epoch core-contract (https://github.com/onflow/flow-core-contracts/pull/397).

- update the core-contract version with the latest epoch contract changes
- node software makes sure that the source of randomness is a hex string of exactly 128 bits and rejects any other string
- add tests for invalid sources

### CODE REVIEW

**Comment by tarakby** [comment]
This PR will be open once https://github.com/onflow/flow-core-contracts/pull/397 is merged

File: model/convert/service_event_test.go [hunk 1]
```diff
@@ -39,6 +39,21 @@ func TestEventConversion(t *testing.T) {
 		},
 	)
 
+	t.Run(
```
[the remark this answers is not in the document]
Comment (tarakby) [reply model/convert/service_event_test.go:42]: https://github.com/onflow/flow-go/pull/5088/commits/3bd56a456cc25e1cb6ed13c47de3452c7fd718f3

**Comment by joshuahannan** [comment]
@tarakby The core contracts PR is merged and is getting upgraded on mainnet tomorrow. Does anything need to be done about this PR?

**Comment by tarakby** [comment]
@joshuahannan the only constraint is to **not** deploy this PR on Mainnet till the contract is upgraded. This is guaranteed once the contract is upgraded so I'll open this PR tomorrow.

**Review by jordanschalm — APPROVED** [review]

**Review by joshuahannan — APPROVED** [review]

File: model/convert/service_event_test.go [hunk 2]
```diff
@@ -39,6 +39,69 @@ func TestEventConversion(t *testing.T) {
...
+			fixture, _ := unittest.EpochSetupFixtureByChainID(chainID)
+			// update the random source in event fixture
+			fixture.Payload = unittest.EpochSetupCCFWithNonHexRandomSource()
+
+			// convert Cadence types to Go types
+			event, err := convert.ServiceEvent(chainID, fixture)
+			require.Error(t, err)
+			require.Nil(t, event)
+		},
+	)
```
[the remark this answers is not in the document]
Comment (tarakby) [reply model/convert/service_event_test.go:74]: good catch

**Pull request merged**